src/Symfony/Component/HttpFoundation/File/MimeType/MimeTypeGuesser.php

```diff
@@ -125,18 +117,40 @@ public function guess($path)
             throw new AccessDeniedException($path);
         }
 
-        if (!$this->guessers) {
+        if (!$guessers = $this->getGuessers()) {
             $msg = 'Unable to guess the mime type as no guessers are available';
             if (!FileinfoMimeTypeGuesser::isSupported()) {
                 $msg .= ' (Did you enable the php_fileinfo extension?)';
             }
             throw new \LogicException($msg);
         }
 
-        foreach ($this->guessers as $guesser) {
+        foreach ($guessers as $guesser) {
             if (null !== $mimeType = $guesser->guess($path)) {
                 return $mimeType;
             }
         }
     }
+
+    /**
+     * @return MimeTypeGuesserInterface[]
+     */
+    private function getGuessers()
+    {
+        if (!$this->nativeGuessersLoaded) {
+            // register all natively provided mime type guessers.
+
+            if (FileinfoMimeTypeGuesser::isSupported()) {
+                $this->guessers[] = new FileinfoMimeTypeGuesser();
```

Comment:
this changes the existing behavior, as they are now registered at the end while the existing logic registers them *before* custom ones.

Reply:
In existing code native guessers are prepended to an empty array in a private constructor. By the time `guess` is called, native loaders are always at the end of the array.

In this PR they are added later. So to have the same behavior, I add them to the end of the array in a reversed order.

Reply:
ah, I forgot the fact that `register` was prepending the guessers